### ISSUE TITLE
UPnP / DLNA support?

### ISSUE THREAD

**Issue by olivermuc** [iina/iina#2173]
Love the player!! 
However, ...any chance UPnP / DNLA support will make it into a future release?

Thanks-
O.

**Comment by Leroy-Melb** [comment]
Yep - only thing preventing me using this daily at the moment. Thanks for the great work.

**Comment by barcia** [comment]
+1

**Comment by stefaanbolle** [comment]
+1

**Comment by hupanqing** [comment]
+10086

**Comment by stefaanbolle** [comment]
This would make vlc useless

**Comment by Shnub** [comment]
For me too, please. :-)

**Comment by Obooman** [comment]
Thanks for the AWESOME project! Are there any decisions been made about this issue?

**Comment by LumberStone** [comment]
+1

**Comment by stefan1983** [comment]
+1 

Any chance to get it implemented?

**Comment by billybones256** [comment]
+1

**Comment by tellidis** [comment]
+1

**Comment by heartbleeded** [comment]
+1

**Comment by stefaanbolle** [comment]
+1

**Comment by qinyang912** [comment]
+1

**Comment by Deusdictum** [comment]
Still +1

**Comment by harshanand120** [comment]
+1. Its the best player for macOS. It is just needs a few features vlc has,then there is no looking back.

**Comment by BenjaminChoou** [comment]
+1. Needs UPnP/DLNA with full function including being a server (cast video to other devices) or a client (being controlled by other devices).

**Comment by CharlieBrown12** [comment]
Please add DLNA!!!!

Its is simply requirement for being a complete video player...

**Comment by adriel** [comment]
I looked into DLNA a few years ago for a different project.. lets just say, there is nothing simple about DLNA from a programming standpoint. At least it wasn't to me.

**Comment by Feivel65** [comment]
I'd also love to be able to access my Homeserver Movies via IINA. Only reason I have the ugly VLC still installed.

**Comment by catap** [comment]
@Feivel65 and music, not only movie!

**Comment by WingedGeek** [comment]
Another request for UPnP; I have an HD HomeRun Prime I use for CATV and VLC can easily find and digest those streams, only thing I'm still using it for... https://imgur.com/a/h3HzefT

**Comment by IvanaXu** [comment]
+1

**Comment by Edlar** [comment]
+1

**Comment by Edlar** [comment]
+1

**Comment by Dentrax** [comment]
Any progress on this? Just download this app to check whether it supports DLNA but couldn't find much on the settings. I wanted to connect my locally hosted DLNA server. Probably going to install VLC if it does not.

cc @lhc70000 @uiryuu

**Comment by kungmo** [comment]
저도 IINA에서 DLNA 지원하는 순간 VLC는 영원히 안 쓸 겁니다.

**Comment by dmitryzrv** [comment]
+1

**Comment by raiden-e** [comment]
+1

**Comment by aklen** [comment]
+1

**Comment by bearnard** [comment]
+1

**Comment by qcq** [comment]
I also found this feature from Elmedia player, I also need this feature to IINA too. Thanks in advance.

**Comment by baptistefkt** [comment]
+1

**Comment by zhengzhu-1m** [comment]
+1

**Comment by Vvorcun** [comment]
+1

**Comment by low-batt** [comment]
There are multiple issues requesting this feature. As this one has the most comments and thumbs up I am going to treat this one as the primary issue for this feature.

There is now a proposed implementation in PR #5877. Currently it fails to build, so it needs more work and of course once it builds and works it will need a careful review.
